Not a valid DAV response

I use [Box](https://www.box.com/). When I call LOCK I get crash "java.io.IOException: Not a valid DAV response". I'm using "sardine-android:0.4".

Response:
```
<?xml version="1.0" encoding="utf-8"?>
<d:prop xmlns:d="DAV:">
  <d:lockdiscovery>
    <d:activelock>
      <d:lockscope>
        <d:exclusive/>
      </d:lockscope>
      <d:locktype>
        <d:write/>
      </d:locktype>
      <d:lockroot>
        <d:href>/dav/ksync/boards/dabea1be-a895-4738-9ee0-d13175ed7ee2.ksync</d:href>
      </d:lockroot>
      <d:depth>infinity</d:depth>
      <d:timeout>Second-</d:timeout>
      <d:locktoken>
        <d:href>opaquelocktoken:c014b505-47e9-4444-93ed-46db4977b569</d:href>
      </d:locktoken>
      <d:owner></d:owner>
    </d:activelock>
  </d:lockdiscovery>
</d:prop>
```

Stack trace:

```
java.io.IOException: Not a valid DAV response
    at com.thegrizzlylabs.sardineandroid.util.SardineUtil.unmarshal(SardineUtil.java:140)
    at com.thegrizzlylabs.sardineandroid.impl.handler.LockResponseHandler.getToken(LockResponseHandler.java:38)
    at com.thegrizzlylabs.sardineandroid.impl.handler.LockResponseHandler.handleResponse(LockResponseHandler.java:27)
    at com.thegrizzlylabs.sardineandroid.impl.handler.LockResponseHandler.handleResponse(LockResponseHandler.java:17)
    at com.thegrizzlylabs.sardineandroid.impl.OkHttpSardine.execute(OkHttpSardine.java:595)
    at com.thegrizzlylabs.sardineandroid.impl.OkHttpSardine.lock(OkHttpSardine.java:406)
    at org.pdapps.kanbani.synclib.server.base.WebDAVServer.lock(WebDAVServer.java:125)
    at org.pdapps.kanbani.synclib.module.SyncModule.lockBoard(SyncModule.java:203)
    at org.pdapps.kanbani.synclib.module.SyncModule.syncBoard(SyncModule.java:164)
    at org.pdapps.kanbani.synclib.module.SyncModule.syncInternal(SyncModule.java:125)
    at org.pdapps.kanbani.synclib.module.SyncModule.access$000(SyncModule.java:36)
    at org.pdapps.kanbani.synclib.module.SyncModule$1.run(SyncModule.java:86)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
    at java.lang.Thread.run(Thread.java:818)
Caused by: org.simpleframework.xml.core.ElementException: Element 'lockroot' does not have a match in class com.thegrizzlylabs.sardineandroid.model.Activelock at line -1
    at org.simpleframework.xml.core.Composite.readElement(Composite.java:527)
    at org.simpleframework.xml.core.Composite.readElements(Composite.java:445)
    at org.simpleframework.xml.core.Composite.access$400(Composite.java:59)
    at org.simpleframework.xml.core.Composite$Builder.read(Composite.java:1383)
    at org.simpleframework.xml.core.Composite.read(Composite.java:201)
    at org.simpleframework.xml.core.Composite.read(Composite.java:148)
    at org.simpleframework.xml.core.Traverser.read(Traverser.java:92)
    at org.simpleframework.xml.core.CompositeInlineList.read(CompositeInlineList.java:190)
    at org.simpleframework.xml.core.CompositeInlineList.read(CompositeInlineList.java:167)
    at org.simpleframework.xml.core.CompositeInlineList.read(CompositeInlineList.java:124)
    at org.simpleframework.xml.core.Composite.readVariable(Composite.java:623)
    at org.simpleframework.xml.core.Composite.readInstance(Composite.java:573)
    at org.simpleframework.xml.core.Composite.readUnion(Composite.java:549)
    at org.simpleframework.xml.core.Composite.readElement(Composite.java:532)
    at org.simpleframework.xml.core.Composite.readElements(Composite.java:445)
    at org.simpleframework.xml.core.Composite.access$400(Composite.java:59)
    at org.simpleframework.xml.core.Composite$Builder.read(Composite.java:1383)
    at org.simpleframework.xml.core.Composite.read(Composite.java:201)
    at org.simpleframework.xml.core.Composite.read(Composite.java:148)
    at org.simpleframework.xml.core.Traverser.read(Traverser.java:92)
    at org.simpleframework.xml.core.Persister.read(Persister.java:625)
    at org.simpleframework.xml.core.Persister.read(Persister.java:606)
    at org.simpleframework.xml.core.Persister.read(Persister.java:584)
    at org.simpleframework.xml.core.Persister.read(Persister.java:480)
    at com.thegrizzlylabs.sardineandroid.util.EntityWithAnyElementConverter.read(EntityWithAnyElementConverter.java:60)
    at com.thegrizzlylabs.sardineandroid.util.EntityWithAnyElementConverter.read(EntityWithAnyElementConverter.java:17)
    at org.simpleframework.xml.convert.RegistryStrategy.read(RegistryStrategy.java:129)
    at org.simpleframework.xml.convert.RegistryStrategy.read(RegistryStrategy.java:108)
    at org.simpleframework.xml.core.Source.getOverride(Source.java:360)
    at org.simpleframework.xml.core.Factory.getConversion(Factory.java:207)
    at org.simpleframework.xml.core.Factory.getOverride(Factory.java:139)
    at org.simpleframework.xml.core.ObjectFactory.getInstance(ObjectFactory.java:63)
    at org.simpleframework.xml.core.Composite.read(Composite.java:139)
    at org.simpleframework.xml.core.Traverser.read(Traverser.java:92)
    at org.simpleframework.xml.core.Persister.read(Persister.java:625)
    at org.simpleframework.xml.core.Persister.read(Persister.java:606)
    at org.simpleframework.xml.core.Persister.read(Persister.java:584)
    at org.simpleframework.xml.core.Persister.read(Persister.java:543)
    at org.simpleframework.xml.core.Persister.read(Persister.java:444)
    at com.thegrizzlylabs.sardineandroid.util.SardineUtil.unmarshal(SardineUtil.java:135)
	... 14 more
```

--- a/src/main/java/com/thegrizzlylabs/sardineandroid/model/Activelock.java
+++ b/src/main/java/com/thegrizzlylabs/sardineandroid/model/Activelock.java
@@ -36,7 +36,7 @@ import org.simpleframework.xml.Root;
  * 
  * 
  */
-@Root
+@Root(strict = false)
 @Namespace(prefix = "D", reference = "DAV:")
 public class Activelock {
 

--- a/src/test/java/com/thegrizzlylabs/sardineandroid/impl/handler/LockResponseHandlerTest.java
+++ b/src/test/java/com/thegrizzlylabs/sardineandroid/impl/handler/LockResponseHandlerTest.java
@@ -1,0 +1,57 @@
+/*
+ * Copyright 2009-2011 Jon Stevens et al.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thegrizzlylabs.sardineandroid.impl.handler;
+
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ */
+public class LockResponseHandlerTest {
+
+    @Test
+    public void testBoxResponse() throws Exception {
+        LockResponseHandler handler = new LockResponseHandler();
+        final String response = "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n" +
+                "<d:prop xmlns:d=\"DAV:\">\n" +
+                "  <d:lockdiscovery>\n" +
+                "    <d:activelock>\n" +
+                "      <d:lockscope>\n" +
+                "        <d:exclusive/>\n" +
+                "      </d:lockscope>\n" +
+                "      <d:locktype>\n" +
+                "        <d:write/>\n" +
+                "      </d:locktype>\n" +
+                "      <d:lockroot>\n" +
+                "        <d:href>fake-lock-root</d:href>\n" +
+                "      </d:lockroot>\n" +
+                "      <d:depth>infinity</d:depth>\n" +
+                "      <d:timeout>Second-</d:timeout>\n" +
+                "      <d:locktoken>\n" +
+                "        <d:href>fake-lock-token</d:href>\n" +
+                "      </d:locktoken>\n" +
+                "      <d:owner></d:owner>\n" +
+                "    </d:activelock>\n" +
+                "  </d:lockdiscovery>\n" +
+                "</d:prop>";
+        final String token = handler.getToken(new ByteArrayInputStream(response.getBytes()));
+        assertEquals("fake-lock-token", token);
+    }
+}

